streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -473,6 +550,21 @@ public final TopologyBuilder addInternalTopic(String topicName) {
         return this;
     }
 
+    /**
+     * Updates the source node containing the Pattern for subscribing
+     * to matching topics with the topic names discovered during partition
+     * assignment.
+     *
+     * @param topics  the topics subscribed to from matching the Pattern given
+     */
+    public synchronized void updatePatternNodeTopics(String[] topics) {
```

Comment:
> Could we not make it public since TopologyBuilder is a public API?

@guozhangwang - I wasn't comfortable with the method being public, but the only point at which topics matching the given pattern are visible to the TopologyBuilder is in the StreamPartitionAssignor class (line 264).  The only other option I came up with was to move StreamPartitionAssignor up out of the internals package and make the method package private, but I don't think that is a good idea.